coderd/workspaceapps.go

```diff
@@ -94,9 +177,30 @@ func (api *API) workspaceAppsProxyPath(rw http.ResponseWriter, r *http.Request)
 		http.Redirect(rw, r, r.URL.String(), http.StatusTemporaryRedirect)
 		return
 	}
-	r.URL.Path = path
 
-	conn, release, err := api.workspaceAgentCache.Acquire(r, agent.ID)
+	r.URL.Path = proxyApp.Path
+	appURL.RawQuery = ""
+
+	proxy := httputil.NewSingleHostReverseProxy(appURL)
+	proxy.ErrorHandler = func(w http.ResponseWriter, r *http.Request, err error) {
+		if proxyApp.DashboardOnError {
```

Comment:
We can't render the dashboard onto a devurls subdomain because we have a PWA app. It will try to make API requests to `app--agent--ws--user.domain.com/api/v2` which will fail. We also don't have a way of redirecting the user to the real access URL and showing an API error page, and we probably don't want to do that either because now if they refresh they'll be on the dashboard and not their devurl.

We'll need to add a small renderable error page similar to v1, but for now I don't think this is worth blocking this PR over. I made a ticket for it: #4040

Reply:
I'm not sure what you mean by PWA app? I don't believe we have one right now.

Reply:
i meant SPA oops

Reply:
The SPA can detect if it's on a subdomain, right? It can just look at the host header and route accordingly.

Reply:
We can't serve the dashboard app from subdomains like these because:

- It's comprised of multiple files (JS, CSS, images) and not just a single HTML file we can return (and these files are inaccessible on devurl subdomains)
- The dashboard assumes the API is at `/api/v2` when it's actually on a different domain as APIs are inaccessible from devurl subdomains
- The dasboard relies on a full session cookie for APIs to work, and we're planning on using a devurl-only session cookie

Reply:
A single HTML file like we use in v1 means we don't need to worry about any of these things. All it needs to be is an error page with the coder logo, an error message, and two buttons

Reply:
Spoke to Kyle on slack about this and we figured it out